File: Common/Math/src/main/java/com/greentree/common/math/vector/FinalVectorNf.java
package com.greentree.common.math.vector;


public class FinalVectorNf extends AbstractVectorNf {
	private static final long serialVersionUID = 1L;

	private final float[] values;

	public FinalVectorNf(float[] values) {
		this.values = values.clone();
	}
	
	public FinalVectorNf(int length) {
		values = new float[length];
	}
	
	@Override
	public float get(int i) {
		return values[i];
	}

	@Override
	public void set(int i, float x) {
		throw new UnsupportedOperationException("change of " + this);
	}

	@Override
	public int size() {
		return values.length;
	}

}

File: Commons/Math/src/main/java/com/greentree/commons/math/vector/FinalVectorNf.kt
package com.greentree.commons.math.vector

class FinalVectorNf(private val values: FloatArray) : AbstractVectorNf {

    override val size
        get() = values.size

    constructor(length: Int) : this(FloatArray(length))

    override fun get(i: Int): Float {
        return values[i]
    }

}